src/type.ts

```diff
@@ -0,0 +1,119 @@
+/**
+ * Defines the structure for configuration options.
+ *
+ * @export
+ * @interface ConfigOption
+ * @property {string} filename - Name/path of the JUnit file.
+ */
+export interface ConfigOption {
+  filename: string;
+}
+
+/**
+ * Represents the structure of a JUnit test suite, detailing its composition and outcomes.
+ *
+ * @export
+ * @interface JUnitTestSuite
+ *
+ * @property {string} name - The name of the JUnit test suite.
+ * @property {number} tests - Total number of tests included in the suite.
+ * @property {number} failures - Count of tests that failed.
+ * @property {number} errors - Number of tests that encountered errors.
+ * @property {number} time - Execution time for the entire test suite, in seconds.
+ * @property {string} timestamp - Date and time of when the test suite was executed (in ISO 8601 format).
+ * @property {string} file - Source code file of this test suite.
+ * @property {JUnitTestSuite[]} testsuites - Optional; array of JUnitTestSuite objects, each representing an individual test suite.
+ * @property {JUnitTestCase[]} testCases - Array of JUnitTestCase objects, each representing an individual test case.
+ * @property {Property[]} [properties] - Optional; array of Property objects for specifying additional metadata related to the test suite.
```

Comment:
Just fyi, you don't need to include the type in these doc comments since you're using TS and the types are defined by the interface itself. Same with the `@export` and `@interface` directives.